cli/internal/fs/package_json.go

```diff
@@ -57,9 +64,77 @@ func ReadPackageJSON(path turbopath.AbsoluteSystemPath) (*PackageJSON, error) {
 	if err != nil {
 		return nil, err
 	}
-	packageJSON := &PackageJSON{}
-	if err := json.Unmarshal(b, packageJSON); err != nil {
+	return UnmarshalPackageJSON(b)
+}
+
+// UnmarshalPackageJSON decodes a byte slice into a PackageJSON struct
+func UnmarshalPackageJSON(data []byte) (*PackageJSON, error) {
+	var rawJSON map[string]interface{}
+	if err := json.Unmarshal(data, &rawJSON); err != nil {
+		return nil, err
+	}
+
+	pkgJSON := &PackageJSON{}
+	if err := json.Unmarshal(data, &pkgJSON); err != nil {
+		return nil, err
+	}
+	pkgJSON.RawJSON = rawJSON
+
+	return pkgJSON, nil
+}
+
+// MarshalPackageJSON Serialize PackageJSON to a slice of bytes
+func MarshalPackageJSON(pkgJSON *PackageJSON) ([]byte, error) {
+	structuredContent, err := json.Marshal(pkgJSON)
+	if err != nil {
+		return nil, err
+	}
+	var structuredFields map[string]interface{}
+	if err := json.Unmarshal(structuredContent, &structuredFields); err != nil {
 		return nil, err
 	}
-	return packageJSON, nil
+
+	fieldsToSerialize := make(map[string]interface{}, len(pkgJSON.RawJSON))
+
+	// copy pkgJSON.RawJSON
+	for key, value := range pkgJSON.RawJSON {
+		fieldsToSerialize[key] = value
+	}
+
+	for key, value := range structuredFields {
+		if isEmpty(value) {
+			delete(fieldsToSerialize, key)
+		} else {
+			fieldsToSerialize[key] = value
+		}
+	}
+
+	var b bytes.Buffer
+	encoder := json.NewEncoder(&b)
+	encoder.SetEscapeHTML(false)
+	encoder.SetIndent("", "  ")
+	if err := encoder.Encode(fieldsToSerialize); err != nil {
+		return nil, err
+	}
+
+	return b.Bytes(), nil
+}
+
+func isEmpty(value interface{}) bool {
+	if value == nil {
```

Comment:
I think a more idiomatic way to structure this would be `switch s := value.(type)`: https://go.dev/tour/methods/16